teos/src/config.rs

```diff
@@ -242,6 +243,22 @@ impl Config {
     pub fn is_default(&self) -> bool {
         self == &Config::default()
     }
+
+    /// Logs non-default options.
+    pub fn log_non_default_options(&self) {
+        let json_default_config = serde_json::json!(&Config::default());
+        let json_config = serde_json::json!(&self);
+
+        for (key, value) in json_config.as_object().unwrap().iter() {
+            if key == "btc_rpc_user" || key == "btc_rpc_password" {
+                continue;
+            }
```

Comment:
Ummm, don't think you wanna censor `btc_rpc_user`!
Is this piece of info confidential?
cc/ @sr-gi

Reply:
> Ummm, don't think you wanna censor `btc_rpc_user`! Is this piece of info confidential? cc/ @sr-gi

Should I remove it or let it stay?

Reply:
Instead of not logging it, log it as `****` as suggested in a previous review.

You can create a temporary variable that is set to `"****"` or `value` depending on whether the key is `btc_rpc_*` and then log using that temp variable instead of `value`

Reply:
You can also do it straightaway in the log macro:

```suggestion
let protected_keys = ["btc_rpc_user", "btc_rpc_password"];

        for (key, value) in json_config.as_object().unwrap().iter() {
            if *value != json_default_config[key] {
                log::info!(
                    "{}: {}",
                    key,
                    if protected_keys.contains(&key.as_str()) {
                        "****".to_owned()
                    } else {
                        value.to_string()
                    }
                );
            }
        }
```

Reply:
Also, notice you have changed the message from what was previously suggested. "Custom config arg:" has been removed.

Reply:
> Also, notice you have changed the message from what was previously suggested. "Custom config arg:" has been removed.

Yes, I've added that now.

Reply:
So looks like bitcoind does hide `rpcuser` so we might be fine doing that as well.
But it also hides `rpcbind`!
Not sure why they choose specific things to be hidden and others not :/
To me, it only makes sense to hide the password.
cc/ @sr-gi

Reply:
You made me do some research about this.

Bitcoin Core does hide arguments that are flagged as `SENSITIVE` [[definition](https://github.com/bitcoin/bitcoin/blob/669af326329b01a739aa5ee79c2e8002818fa4cc/src/common/args.h)][[settings](https://github.com/bitcoin/bitcoin/blob/669af326329b01a739aa5ee79c2e8002818fa4cc/src/init.cpp)]. This, **currently**, includes: `rpcauth`, `rpcuser`, `rpcpassword` and `torpassword`. `rpcbind` used to be part of this but it was recently dropped (so it won't be from 25.0 onwards): https://github.com/bitcoin/bitcoin/commit/b9d567454159f062ce84353f5821d6e6daf433bd